### PR TITLE
A number of changes, mainly in data schema.

### PR DESCRIPTION
**Pull request by mzhurovich**

### CODE REVIEW

File: schema.h [hunk 1]
```diff
@@ -35,11 +35,26 @@ SOFTWARE.
 // Data structures for internal storage.
 enum class UID : uint64_t { INVALID = 0u };
 enum class CID : uint64_t { INVALID = 0u };
-enum class ANSWER : int { UNSEEN = 0, CTFO = 1, TFU = 2, TIFB = -1 };
+enum class ANSWER : int { UNSEEN = 0, CTFO = 1, TFU = 2, SKIP = -1 };
+enum class AUTH_TYPE : int { UNDEFINED = 0, IOS };
+
+const std::vector<unsigned int> LEVEL_SCORES{
```
Comment (dkorolev) [inline schema.h:41]: This won't pass the ODR and `make check`. If it should be in a header file, wrap it into a `Singleton`.
Better yet, have this singleton read and parse an input JSON with the mapping.

File: schema.h [hunk 2]
```diff
@@ -35,11 +35,26 @@ SOFTWARE.
...
+    480000,   // "Dolphin"
+    960000,   // "Elephant"
+    1920000,  // "Chimp"
+    3840000   // "Skrik"
+};
 
 struct User : yoda::Padawan {
   UID uid = UID::INVALID;
-  uint64_t score = 0u;
+  unsigned int level = 0u;  // User level [0, 9].
```
Comment (dkorolev) [inline schema.h]: uint8_t is probably cleaner here.

File: schema.h [hunk 3]
```diff
@@ -47,53 +62,77 @@ struct User : yoda::Padawan {
...
+struct AuthKey {
+  std::string key1 = "";
+  std::string key2 = "";
+  AUTH_TYPE type = AUTH_TYPE::UNDEFINED;
+
+  AuthKey() = default;
+  AuthKey(const std::string& key1, const std::string& key2, AUTH_TYPE type)
+      : key1(key1), key2(key2), type(type) {}
+  size_t Hash() const { return std::hash<std::string>()(key1) ^ std::hash<std::string>()(key2); }
+  bool operator==(const AuthKey& rhs) const { return key1 == rhs.key1 && key2 == rhs.key2; }
```
Comment (dkorolev) [inline schema.h]: While I see the reason to not include `type` in `Hash()`, it should certainly be here in `operator==`.

(In `Hash()`, it's OK to only keep the random ID, if you ask me.)

File: schema.h [hunk 4]
```diff
@@ -47,53 +62,77 @@ struct User : yoda::Padawan {
...
+// AuthKey structure defines generic authentication key, which consists of a pair of `std::string`s.
+// `key1` and `key2` may be used in different ways, depending on the authentication type.
+// For `type = AUTH_TYPE::IOS`:
+//  - key1 = Device ID
+//  - key2 = Application key, a random number, generated once in a lifetime in iOS application on its first
+//  launch.
+struct AuthKey {
+  std::string key1 = "";
+  std::string key2 = "";
+  AUTH_TYPE type = AUTH_TYPE::UNDEFINED;
```
Comment (dkorolev) [inline schema.h:78]: I'd move `type` to the top, and rename `key1`, `key2`, into `did`, and `did` (`device_key`, and `random_key`).

Comment (dkorolev) [inline schema.h:78]: Disregard it :+1:

**Comment by dkorolev** [comment]
@mzhurovich : Very minor things. Ready to merge as you are!

File: schema.h (hunk 1 above)
Comment (mzhurovich) [inline schema.h:41]: How does it pass `make check`?

Comment (dkorolev) [inline schema.h:41]: OMFG, it really does! Starts failing it only if `LEVEL_SCORES` is made non-const. Even passes if I take a pointer to the `const` one.

Clean test: `ln -sf schema.h 1.cc && ln -sf schema.h 2.cc && clang++ -std=c++11 -c 1.cc && clang++ -std=c++11 -c 2.cc && clang++ -o 12 1.o 2.o`

Looks like this StackOverflow answer is right: `"const" implies internal linkage`.
(http://stackoverflow.com/questions/2268749/defining-global-constant-in-c)

Thanks!
Dima

Comment (mzhurovich) [inline schema.h:41]: Cool!
Let's leave it as is in this PR.

Comment (dkorolev) [inline schema.h:41]: Yep, +1.

**Comment by mzhurovich** [comment]
@dkorolev Thanks! Made a couple of small changes after your comments, PTAL.

**Comment by dkorolev** [comment]
LGTM! Merging.

**Pull request merged**